File: week-2/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string string = get_string("What is the string to analyze? \n");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int i = 0, stringLength = strlen(string); i < stringLength + 1; i++)
    {
        if ((string[i] >= 'a' && string[i] <= 'z') || (string[i] >= 'A' && string[i] <= 'Z'))
        {
            letters++;
        }
        if (string[i] == ' ' || string[i] == '\0')
        {
            words++;
        }
        if (string[i] == '!' || string[i] == '.' || string[i] == '?')
        {
            sentences++;
        }
    }

    float averageWordsPer100 = (100 / (float) words) * (float) letters;
    float averageSentencePer100 = (100 / (float) words) * (float) sentences;
    int readingIndex = round(0.0588 * averageWordsPer100 - 0.296 * averageSentencePer100 - 15.8);

    if (readingIndex < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (readingIndex > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", readingIndex);
    }

    return 0;
}

File: week-6/problem-set/sentimental-readability/readability.py
from cs50 import get_string


text = get_string("Text: ")

i = len(text)

letters = 0
words = 1
sentences = 0


for x in range(i):
    c = text[x]
    if c.isalpha():
        letters += 1


    if c == " ":
        words += 1


    if c in [".", "!", "?"]:
        sentences += 1

# Calculation
L = (letters / words) * 100
s = (sentences / words) * 100
subindex = 0.0588 * L - 0.296 * s - 15.8
index = round(subindex)
if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")


#back here
